### PR TITLE
[RLlib] Change scope of PPO tests to comply with currently supported set of spaces in RLModules for PPO

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

This PR changes the scope of the PPO test_supported_spaces test.
As we not yet support all previously supported spaces, this PR makes it so that we skip the tests that we don't "want".

It also restructures the tests:
- Test all observation spaces with a default action space "discrete"
- Test all actions spaces with a default observations space "discrete"
(whereas before all possible spaces on both sides where zipped to create test cases)
- Separate more tested algos from each other to make up for the 2x runtime of the above change

### CODE REVIEW

File: rllib/core/models/base.py [hunk 1]
```diff
@@ -195,7 +202,8 @@ class Encoder(Model, abc.ABC):
     Similarly, their output_spec contains the latent space dimensions.
     Encoders can be recurrent, in which case the state should be part of input- and
     output_specs. The latents that are produced by an encoder are fed into subsequent
-    heads.
+    heads. Any implementation of Encoder should also be callable. This should be done
+    by also inheriting from a framework-specific model base-class, s.a. TorchModel.
```
Comment (ArturNiederfahrenhorst) [inline rllib/core/models/base.py:206]: This is more feedback from Jun - we were missing this hint.

**Review by kouroshHakha — CHANGES_REQUESTED** [review]
dude this is super hacky, and not approvable.
My suggestion (the general sketch):
```
LEARNER_ENABLED_OBS_SPACES = [...]
LEARNER_ENABLED_ACTION_SPACES = [...]
def _do_check(algo, config, a_name, o_name):
     config_copy = config.copy()
     config_copy.validate()
     if config_copy.enable_api:
              obs_space_list = LEARNER_ENABLED_OBS_SPACES
              act_space_list = LEARNER_ENABLED_ACTION_SPACES
     else: ...
     
     if o_name not in obs_space_list and in THE_OLD_OBS_LIST:
            warn(message)
            return
      if a_name not in act_space_list and in THE_OLD_ACTION_LIST:
            warn(message)
            return 
```

**Comment by ArturNiederfahrenhorst** [comment]
@kouroshHakha I understand that it looks hacky. But the code you propose makes it so that hitting an unsupported action space OR and unsupported observations space simply skips the test. Thereby omitting tests that would have a supported space.

**Comment by ArturNiederfahrenhorst** [comment]
This test works by zipping the spaces together. I don't think it is worth restructuring the rest of the test around your code.

**Comment by ArturNiederfahrenhorst** [comment]
As per our offline discussion, we are restructuring the test itself.

File: rllib/tests/test_supported_spaces.py [hunk 2]
```diff
@@ -60,12 +61,43 @@
...
+
+DEFAULT_OBSERVATION_SPACE = DEFAULT_ACTION_SPACE = "discrete"
+
 
 def check_support(alg, config, train=True, check_bounds=False, tf2=False):
     config["log_level"] = "ERROR"
     config["env"] = RandomEnv
 
     def _do_check(alg, config, a_name, o_name):
+        if alg == "PPO":
```
Comment (kouroshHakha) [inline rllib/tests/test_supported_spaces.py:78]: We decided to not have this algo == "PPO" specialization, right? 
We should check for which obs / action space to choose based on whether the enable_rl_module_api is set or not.

File: rllib/tests/test_supported_spaces.py [hunk 3]
```diff
@@ -26,12 +27,12 @@
...
 from ray.rllib.utils.test_utils import framework_iterator
 
+logger = logging.getLogger(__name__)
+
 ACTION_SPACES_TO_TEST = {
     # Test discrete twice here until we support multi_binary action spaces
-    "discrete": Discrete(5),  # TODO: (Artur) Support "multi_binary"
-    "discrete2": Discrete(5),
+    "discrete": Discrete(5),
     "vector1d": Box(-1.0, 1.0, (5,), dtype=np.float32),
```
Comment (kouroshHakha) [inline rllib/tests/test_supported_spaces.py:35]: rename: vector1d --> continuous

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**